tests/test_vector_linz_in_bounds/test_vector_linz_in_bounds.py

```diff
@@ -0,0 +1,159 @@
+            shutil.rmtree(cls.cache_dir)
+        cls.cache_dir.mkdir()
+
+        # create fake catchment boundary
+        x0 = 1477354
+        x1 = 1484656
+        y0 = 5374408
+        y1 = 5383411
+        catchment = shapely.geometry.Polygon([(x0, y0), (x0, y1), (x1, y1), (x1, y0)])
+        catchment = geopandas.GeoSeries([catchment])
+        catchment = catchment.set_crs(cls.instructions['instructions']['projection'])
+
+        # save faked catchment file
+        catchment_dir = cls.cache_dir / "catchment"
+        catchment.to_file(catchment_dir)
+        shutil.make_archive(base_name=catchment_dir, format='zip', root_dir=catchment_dir)
+        shutil.rmtree(catchment_dir)
+
+        # cconvert catchment file to zipfile
+        catchment_dir = pathlib.Path(str(catchment_dir) + ".zip")
+        catchment_polygon = geopandas.read_file(catchment_dir)
+
+        # Run pipeline - download files
+        cls.runner = vector.Linz(cls.instructions['instructions']['apis']['linz']['key'], crs=None,
+                                 bounding_polygon=catchment_polygon, verbose=True)
+
+    @classmethod
+    def tearDownClass(cls):
+        """ Remove created cache directory. """
+
+        if cls.cache_dir.exists():
+            shutil.rmtree(cls.cache_dir)
+
+    def compare_to_benchmark(self, features: geopandas.GeoDataFrame, benchmark: dict, description: str,
+                             column_name: str):
+        """ Compare the features attributes (total area, total length, columns, first five values of a column) against
+        those recorded in a benchmark. """
+
+        # check various shape attributes match those expected
+        self.assertEqual(features.loc[0].geometry.geometryType(), benchmark['geometryType'], "The geometryType of the" +
+                         f" returned {description} `{features.loc[0].geometry.geometryType()}` differs from the " +
+                         f"expected {benchmark['geometryType']}")
+        self.assertEqual(list(features.columns), benchmark['columns'], f"The columns of the returned {description}" +
+                         f" lines `{list(features.columns)}` differ from the expected {benchmark['columns']}")
+        self.assertEqual(list(features[column_name][0:5]), benchmark[column_name], "The value of the 'id' column for " +
+                         f"the first five entries `{list(features[column_name][0:5])}` differ from the expected " +
+                         f"{benchmark[column_name]}")
+        self.assertEqual(features.geometry.area.sum(), benchmark['area'], f"The area of the returned {description}" +
+                         f"`{features.geometry.area.sum()}` differs from the expected {benchmark['area']}")
+        self.assertEqual(features.geometry.length.sum(), benchmark['length'], "The length of the returned " +
+                         f"{description} `{features.geometry.length.sum()}` differs from the expected " +
+                         "{benchmark['length']}")
```

Comment:
This is a nice extracted method. Easy to use and easy to understand

Reply:
Thanks! I was stocked when I made the change. It was becoming a nightmare to keep them all up to date!